### PR TITLE
Validation and model specs

### PR DESCRIPTION
**Pull request by diegoyon**
**Validation and model specs**
Added the following validations
For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

Added unit tests for all models' methods and validations

### CODE REVIEW

File: app/models/author.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
-    posts.last(3)
-  end
-end
+class Author < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  has_many :posts
+
+  validates :name, presence: true
+  validates :posts_counter, comparison: { greater_than_or_equal_to: 0 }
```
Comment (muneebulrehman) [inline app/models/author.rb]: Kindly also add the `only_integer` as a validation for posts_counter so as to accept only integers.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
+  after_save :update_posts_counter
+
+  belongs_to :author
+  has_many :comments
+  has_many :likes
+
+  validates :title, presence: true
+  validates :title, length: { maximum: 250 }
+  validates :comments_counter, comparison: { greater_than_or_equal_to: 0 }
+  validates :likes_counter, comparison: { greater_than_or_equal_to: 0 }
```
Comment (muneebulrehman) [inline app/models/post.rb]: Kindly also add the `only_integer` as a validation for comments_counter and likes_counter so as to accept only integers.

File: app/models/post.rb [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
-end
+class Post < ApplicationRecord
+  after_save :update_posts_counter
+
+  belongs_to :author
+  has_many :comments
+  has_many :likes
+
+  validates :title, presence: true
+  validates :title, length: { maximum: 250 }
```
Comment (muneebulrehman) [inline app/models/post.rb]: - Optional
You can combine these two validations into one. Like this ⬇️
`validates : title, presence: true, length: {maximum: 250}`

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @diegoyon , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- All tests passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @diegoyon , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**